### PR TITLE
[android][statusbar] Fix StatusBar background color for light appearance

### LINKED ISSUE #7298: Default status bar not visible on Android
More discussion in Slack: https://exponent-internal.slack.com/archives/C1QNF5L3C/p1583535485017100

### PR DESCRIPTION
**Pull request by bbarthec**
# Why

- Resolves #7298
- Followup of #6506
- fixed wrongly parsed RGBA (Android expects ARGB format)

# How

- `Home` is basing on `expo-device` regarding `light-content` status bar content on pre Android 6.0, but `expo-device` wasn't included in `HomeExperience` (introduces here: #6506) - fixed by adding `DevicePackage` in `HomeExperience`.
- Upon having `light-content` status bar style (default value on Android) and not selecting any `backgroundColor` (default is `transparent`) icons in status bar weren't visible at all in template that is shipped with white background (they were blending in) - added fallback for `light-content` when no `backgroundColor` is selected - see changes in docs.

# Changes in pictures

## Home

| published ExpoClient | master | this PR |
|:-----------------:|:------:|:-------:|
| android_sdk36_client | android_master_client | android_PR_client |

## Launched Experience (default StatusBar)

Status Bar looks exactly the same during LoadingScreen phase.

| published ExpoClient | master | this PR |
|:-----------------:|:------:|:-------:|
| android_sdk36_experience_launched | android_master_experience_launched | android_PR_experience_launched |

# TODOs after merge
- [x] cherry-pick to `sdk-37`
- [ ] ensure default configuration works in `standalone` too

### CODE REVIEW

File: docs/pages/versions/unversioned/guides/configuring-statusbar.md [hunk 1]
```diff
@@ -115,7 +115,7 @@ Example:
     "hidden": false, // default value
     "translucent": true, // default value to align with default iOS behavior
     "barStyle": "light-content", // default value
-    "backgroundColor": "#00000000" // default value - transparent color
+    "backgroundColor": "#00000000" // default value depends on "barStyle" value
```
Comment (brentvatne) [inline docs/pages/versions/unversioned/guides/configuring-statusbar.md]:
```suggestion
    "backgroundColor": "#00000000" // default value depends on "barStyle" value - fully-transparent when it is `dark-content` and semi-transparent black for `light-content` 
```

File: android/expoview/src/main/java/host/exp/exponent/utils/ExperienceActivityUtils.java [hunk 2]
```diff
@@ -134,19 +136,43 @@ public static void configureStatusBar(@NonNull JSONObject manifest, final Activi
...
 
-      if (statusBarStyle != null) {
-        setStyle(statusBarStyle, activity);
+      // Color passed from manifest is in format '#RRGGBB(AA)' and Android uses '#AARRGGBB'
+      String normalizedStatusBarBackgroundColor = RGBAtoARGB(statusBarBackgroundColor);
+
+      if (normalizedStatusBarBackgroundColor == null || !ColorParser.isValid(normalizedStatusBarBackgroundColor)) {
+        // backgroundColor is invalid or not set
+        if (appliedStatusBarStyle.equals(STATUS_BAR_STYLE_LIGHT_CONTENT)) {
+          // appliedStatusBarStyle is "white-content" so background color should be semi transparent black
```
Comment (brentvatne) [inline android/expoview/src/main/java/host/exp/exponent/utils/ExperienceActivityUtils.java]:
```suggestion
          // appliedStatusBarStyle is "light-content" so background color should be semi transparent black
```

File: android/expoview/src/main/java/host/exp/exponent/utils/ExperienceActivityUtils.java [hunk 3]
```diff
@@ -134,19 +136,43 @@ public static void configureStatusBar(@NonNull JSONObject manifest, final Activi
...
+        appliedStatusBarStyle = setStyle(statusBarStyle, activity);
       }
 
-      if (statusBarStyle != null) {
-        setStyle(statusBarStyle, activity);
+      // Color passed from manifest is in format '#RRGGBB(AA)' and Android uses '#AARRGGBB'
+      String normalizedStatusBarBackgroundColor = RGBAtoARGB(statusBarBackgroundColor);
+
+      if (normalizedStatusBarBackgroundColor == null || !ColorParser.isValid(normalizedStatusBarBackgroundColor)) {
+        // backgroundColor is invalid or not set
```
Comment (brentvatne) [inline android/expoview/src/main/java/host/exp/exponent/utils/ExperienceActivityUtils.java]: @bbarthec - can you ensure that expo-cli will warn people if the given background color is invalid?

**Review by brentvatne — APPROVED** [review]

File: android/expoview/src/main/java/host/exp/exponent/utils/ExperienceActivityUtils.java (hunk 3 above)
Comment (bbarthec) [reply android/expoview/src/main/java/host/exp/exponent/utils/ExperienceActivityUtils.java]: Opened new issue for this one https://github.com/expo/expo/issues/7322

**Pull request merged**